### PR TITLE
stable version ready to be released (10.30.17)

### PR DESCRIPTION
**Pull request by AntonRodenhauser**

### CODE REVIEW

File: SOURCE/Fiducializing/autoProcessSignal.m [hunk 1]
```diff
@@ -111,29 +156,30 @@
 % determine the beats, where autoprocessing didn't quite work ( eg those with very high variance)
 % fill AUTOPROCESSING.faultyBeatInfo and AUTOPROCESSING.faultyBeatIndeces with info
 
-global AUTOPROCESSING
+global AUTOPROCESSING ScriptData
```
Comment (jessdtate) [inline SOURCE/Fiducializing/autoProcessSignal.m]: Why didn't you follow the all caps global variable convention?  You should

File: SOURCE/Fiducializing/autoProcessSignal.m [hunk 2]
```diff
@@ -165,56 +211,80 @@
 global TS ScriptData AUTOPROCESSING
 
 
+global times
```
Comment (jessdtate) [inline SOURCE/Fiducializing/autoProcessSignal.m]: Same as before

File: SOURCE/Fiducializing/autoProcessSignal.m [hunk 3]
```diff
@@ -367,105 +484,59 @@ function DetectRecovery(newBeatIdx)
...
-
-if abs(c(1)) < 100*eps, dx = 0; else dx = -c(2)/(2*c(1)); end
-
-dvdt = 2*c(1)*dx+c(2);
-
-dx = median([-0.5 dx 0.5]);
-
-x = mi+dx-1;
 
+success = 1;
```
Comment (jessdtate) [inline SOURCE/Fiducializing/autoProcessSignal.m]: Consider making this variable name different from all the other times that you use this

File: SOURCE/Fiducializing/plotAutoProcFids.m [hunk 4]
```diff
@@ -91,7 +117,7 @@ function SetupNavigationBar(handle)
 
 function initDisplayButtons(fig)
 % initialize everything in figure exept the plotting stuff.. 
-global AUTOPROCESSING
+global AUTOPROCESSING ScriptData
```
Comment (jessdtate) [inline SOURCE/Fiducializing/plotAutoProcFids.m]: Same as above

File: SOURCE/Fiducializing/plotAutoProcFids.m [hunk 5]
```diff
@@ -877,7 +903,7 @@ function UpdateSlider(handle,~)
 
 function DisplayButton(cbobj)
 %callback function to all the buttons
-global ScriptData AUTOPROCESSING
+global ScriptData
```
Comment (jessdtate) [inline SOURCE/Fiducializing/plotAutoProcFids.m]: and here

File: SOURCE/Fiducializing/plotAutoProcFids.m [hunk 6]
```diff
@@ -965,7 +991,7 @@ function InitFiducials(fig)
 function FidsToEvents
 %puts .allFids into .EVENTS
 
-global TS ScriptData AUTOPROCESSING;
+global ScriptData AUTOPROCESSING;
```
Comment (jessdtate) [inline SOURCE/Fiducializing/plotAutoProcFids.m]: again

File: SOURCE/Functions/fids/fidsIntegralMap.m [hunk 7]
```diff
@@ -22,6 +22,9 @@
 %
 %
 
+global aaa
```
Comment (jessdtate) [inline SOURCE/Functions/fids/fidsIntegralMap.m]: This variable could be more descriptive and all caps

File: SOURCE/PFEIFER.m [hunk 8]
```diff
@@ -139,19 +165,19 @@ function saveProcessingData()
...
 
-function updateFigure(handle)
+function updateFigure(figObj)
 % changes all Settings in the figure ( that belongs to handle) according to
 % ScriptData.  
 %Updates everything in the gui figures, including File Listbox etc..
 % handle is gui figure object
-    
+
 global ScriptData;
```
Comment (jessdtate) [inline SOURCE/PFEIFER.m]: here again

File: SOURCE/PFEIFER.m [hunk 9]
```diff
@@ -421,6 +464,57 @@ function CloseFcn(~)
 clear global ProcessingData ScriptData FIDSDISPLAY SLICEDISPLAY TS
 end
 
+
+
+function setupToolSelectionDropdownmenus(figObj)
+global ScriptData
```
Comment (jessdtate) [inline SOURCE/PFEIFER.m]: and here

File: SOURCE/PFEIFER.m [hunk 10]
```diff
@@ -1166,45 +1307,6 @@ function KeyPress(handle)
...
-%       Export User Settings
-% - split current ts into n_groups sub-ts structures, delete original ts
-% - Do_interpolation again, for each sub-ts
-% - SAVE the ts structures, 
-% - do integral maps, if selected and save them
-% - do Activation maps, if that option is selected
-
 success = 0;
 olddir = pwd;
 global ScriptData TS
```
Comment (jessdtate) [inline SOURCE/PFEIFER.m]: and here

File: SOURCE/TOOLS/act_detection/ActDetect.m [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+% AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+% LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+% OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+% SOFTWARE.
+
+
+
+
+
+function x = ActDetect(sig,win,deg)
```
Comment (jessdtate) [inline SOURCE/TOOLS/act_detection/ActDetect.m]: variable names in this function could be better

File: UNIT_TESTING/basic_unit_test.m [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+    executeCallback(settingsFigure,'DATAFILE',pathToProcDataFile)
+    executeCallback(settingsFigure,'SCRIPTFILE',pathToScriptDataFile)
+catch
+    disp('-----UNIT TEST: basic_unit_test --------')
+    disp('ERROR: Could not load helper files')
+end
+
+
+%%%% set all the paths
+global ScriptData
```
Comment (jessdtate) [inline UNIT_TESTING/basic_unit_test.m]: here again

**Review by jessdtate — CHANGES_REQUESTED** [review]

**Comment by jessdtate** [comment]
Some of the files are still missing license info.  This doesn't have to be added on this PR, but it does need to be added.

**Comment by jessdtate** [comment]
The PDF needs to be removed from the PR. I will put it in Devbuilds

**Comment by bzenger** [comment]
Hi Jess, 

Can you clarify? The JOSS requirements is just a license file in the repository. For protection purposes should we put that in the preamble of every file? 

Brian

**Comment by jessdtate** [comment]
It should be on every file of code as a standard practice.

**Comment by jessdtate** [comment]
You have to merge the master branch now

**Review by bzenger — APPROVED** [review]
Should be good to me.

**Review by jessdtate — APPROVED** [review]

**Pull request merged**